std/uni/package.d

```diff
@@ -3373,7 +3373,7 @@ private:
         }
         else
             SP.destroy(data);
-        assert(!data.ptr);
+        assert(!&data[0]);
```

Comment:
This cause a range violation.
```suggestion
        assert((() => !data.ptr)());
```

Reply:
Wait what ? Something is off. `!data.ptr` should be safe.

Reply:
Isn't acessing `.ptr` always un-`@safe` by convention? (Allthough it cannot really do unsafe stuff)

Reply:
E.g. `assert(arr.ptr)` is rejected.

Reply:
I guess ideally the compiler should allow `@safe` `.ptr` access for boolean testing right like in this case?

Reply:
Accessing the content *pointer to* by ptr is unsafe. Checking that the pointer itself is not null is completely `@safe`.
E.g. the following compiles fine for me:
```D
void main () @safe
{
    int[] arr;
    assert(!arr.ptr);
}
```

But if you try to assign it to a variable you'll get an error.

Reply:
> Accessing the content _pointer to_ by ptr is unsafe. Checking that the pointer itself is not null is completely `@safe`.
> E.g. the following compiles fine for me:

As describe below the problem is the call _d_assert_fail!"!"(arr.ptr).

Reply:
> Accessing the content _pointer to_ by ptr is unsafe. Checking that the pointer itself is not null is completely `@safe`.

Agreed but I wasn't sure because the following doesn't compile:
```d
void main () @safe
{
    int[] arr;
    assert(arr.ptr); // no !
}
```